neuro_data/static_images/dataset_config.py

```diff
@@ -63,6 +71,91 @@ def input(self, scan_key):
             frame = frame[:, None, ...]
             return trial_idx, cond, frame, np.full(len(trial_idx), "stimulus.Frame")
 
+    class NeuroStaticValidFrame(dj.Part):
+        # only load stimulus_type=='stimulus.Frame'
+        definition = """
+        -> master
+        ---
+        -> Preprocessing
+        """
+        content = [
+            {"preproc_id": 9},
+            {"preproc_id": 0},
+        ]
+
+        def input(self, scan_key):
+            """Load all frames included in InputResponse.Input and filtered by `valid` in Eye and Treadmill"""
+            params = (self * Preprocessing).fetch1()
+            if not (params["gamma"] or params["linear_mon"]):
+                trial_idx, cond, frame = (
+                    InputResponse.Input * Frame & scan_key & params
+                ).fetch(
+                    "trial_idx",
+                    "condition_hash",
+                    "frame",
+                    order_by="row_id",  # order by row_id to ensure the order matches Eye and Treadmill
+                )
+                valid_eye = (Eye & scan_key & params).fetch1("valid")
+                valid_treadmill = (Treadmill & scan_key & params).fetch1("valid")
+                valid = valid_eye & valid_treadmill
+                # keep only valid trials
+                trial_idx = trial_idx[valid]
+                cond = cond[valid]
+                frame = np.stack(frame)[valid]
+                # check if all frames have the same shape
+                assert (
+                    len(frame.shape) == 3
+                    and frame.shape[1] == params["row"]
+                    and frame.shape[2] == params["col"]
+                ), "dimension mismatch, only support 3-D frame (B,H,W)"
+                # adjust frame shape to (B,1,W,H)
+                frame = frame[:, None, ...]
+            else:
+                raise NotImplementedError(
+                    f'InputConfig: gamma={params["gamma"]}, linear_mon={params["linear_mon"]} not implemented!'
+                )
+            return trial_idx, cond, frame, np.full(len(trial_idx), "stimulus.Frame")
```

Comment:
remove hard-coded stimulus_type.

Reply:
`stimulus_type` is now fetched from `stimulus.Condition`